### PR TITLE
Upgrade controller-runtime dependency to latest (v0.10.3)

### PR DESCRIPTION
**Pull request by ptnapoleon**
## What does this PR do?

- [ ] Adds new functionality
- [ ] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- This is needed for other features that we plan to add. This ended up requiring a change to the e2e-tests, so I thought I'd split it out as a separate PR

## Code Quality Checklist

- [ ] The documentation is up to date.
- [ ] My code is sufficiently commented and passes continuous integration checks.
- [ ] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [ ] I manually tested the following steps:
    - `x`
    - [ ] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,7 +20,7 @@ test: generate manifests
 	go test $(shell go list ./... | grep -v chaos-controller/controllers) -coverprofile cover.out
 
 # Run e2e tests (against a real cluster)
-e2e-test: generate manifests
+e2e-test: generate install
```
Comment (ptnapoleon) [inline Makefile:23]: This is the big change to debate. `install` is now required for e2e-test to work. We sort of always ran it first anyway though, but separately?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 ---
 variables:
-  CURRENT_CI_IMAGE: 0.4.0
+  CURRENT_CI_IMAGE: 0.4.1
```
Comment (ptnapoleon) [inline .gitlab-ci.yml:8]: I have no idea why building on gitlab didn't work until I rebuilt the CI image

**Review by Azoam — APPROVED** [review]
LGTM, I have nothing to say regarding the replacing of manifests to install for the e2e-test if that is required for the tests to run in the upgraded version.

File: Makefile (hunk 1 above)
Comment (Devatoria) [reply Makefile:23]: I think it is an acceptable change.

**Review by Devatoria — APPROVED** [review]

**Pull request merged**